rpc/handlers.go

```diff
@@ -110,17 +111,45 @@ func (h *Handler) BlockWithTxHashes(id BlockID) (*BlockWithTxHashes, *jsonrpc.Er
 		txnHashes[index] = txn.Hash()
 	}
 
+	l1H, jsonErr := h.l1Head()
+	if jsonErr != nil {
+		return nil, jsonErr
+	}
+
 	status := StatusAcceptedL2
-	if id.Pending {
+	if block.Hash != nil && isL1Verified(block.Number, l1H) {
+		status = StatusAcceptedL1
+	} else if id.Pending {
```

Comment:
I
```suggestion
	if id.Pending {
		status = StatusPending
	} else if isL1Verified(block.Number, l1H) {
```

we dont need block.Hash nil-check this way